Console/UI/Pages/EventsLog.xaml.cs

```diff
@@ -126,94 +140,114 @@ public EventsLog()
-        }
-
-        private void EventsLog_Unloaded(object sender, RoutedEventArgs e)
-        {
-            securityLog.EntryWritten -= SecurityLog_EntryWritten;
-        }
-
-        private void EventsLog_Loaded(object sender, RoutedEventArgs e)
+        private void StartHandlingSecurityLogEvents()
         {
             try
             {
-                securityLog = new EventLog("security");
+                if (securityLog is null)
+                {
+                    securityLog = new EventLog("security");
+                }
+                securityLog.BeginInit();
+                Dispatcher.Invoke(() => { ScanProgressMax = securityLog.Entries.Count; });
+                eventsLogFilters.ResetTcpFilter();
+                SecurityLogEntryIndex_last = securityLog.Entries.Count - 1;
+                _ = Task.Run(() => { InitEventLog(SecurityLogEntryIndex_last, value => ScanProgress = value); }).ConfigureAwait(false);
                 securityLog.EntryWritten += SecurityLog_EntryWritten;
                 securityLog.EnableRaisingEvents = true;
+                securityLog.EndInit();
+
             }
             catch (Exception exc)
             {
                 LogHelper.Error("Unable to connect to the event log", exc);
                 throw;
             }
 
-            Task.Run(() => InitEventLog(value => ScanProgress = value));
+        }
+        private void PauseHandlingSecurityLogEvents()
+        {
+            securityLog.EnableRaisingEvents = false;
+        }
+        private void StopHandlingSecurityLogEvents()
+        {
+            PauseHandlingSecurityLogEvents();
+            securityLog?.Dispose();
         }
 
         private void SecurityLog_EntryWritten(object sender, EntryWrittenEventArgs e)
         {
+            //LogHelper.Debug($"EntryWritten event ...");
             if (!FirewallHelper.IsEventAccepted(e.Entry))
             {
                 return;
             }
 
-            LogEntryViewModel entry = EntryViewFromLogEntry(e.Entry);
-            if (entry != null)
+            lock (EntriesLock)
             {
-                Dispatcher.Invoke(() =>
+                if (e.Entry != null && e.Entry.Index > SecurityLogEntryIndex_last)
                 {
-                    if (!TargetCollectionNewEntries.Contains(entry, EntryComparer.Instance))
+                    LogEntryViewModel entry = EntryViewFromLogEntry(e.Entry);
+                    if (entry != null)
                     {
-                        TargetCollectionNewEntries.Add(entry);
+                        Dispatcher.Invoke(() =>
+                        {
+                            Dispatcher.Invoke(() => { LogEntries.Add(entry); ScanProgress = LogEntries.Count; });
+                        });
                     }
-                });
+                }
+                if (ScanProgressMax < LogEntries.Count)
+                {
+                    // overflow handling
+                    LogEntries.RemoveAt(0);

```

Comment:
This is.... awkward? ScanProgressMax should never be greater than LogEntries.Count, does it?

Reply:
Yes the LogEntries did grow forever if we just add events to it 😏

Reply:
Oops my bad. I guess I forgot using my brain before writing that comment :-D